front/cypress/e2e/admin/users/seat-billing.cy.ts

```diff
@@ -0,0 +1,708 @@
+      cy.apiGetAppConfiguration().then((appConfigurationResponse) => {
+        let additionalAdmins =
+          appConfigurationResponse.body?.data.attributes.settings.core
+            .additional_admins_number;
+        let maximumAdmins =
+          appConfigurationResponse.body?.data.attributes.settings.core
+            .maximum_admins_number;
+
+        cy.apiGetSeats().then((seatsResponse) => {
+          let usedSeats = seatsResponse.body.data.attributes.admins_number;
+          let totalSeats = additionalAdmins + maximumAdmins;
+          let remainingSeats = totalSeats - usedSeats;
+
+          cy.visit('/admin/users/admins-managers');
+          cy.acceptCookies();
+          cy.get('[data-cy="e2e-admin-and-moderator-count"]').contains(
+            `${adminAndmoderatorsCount}`
+          );
+          cy.get('[data-cy="e2e-admin-remaining-seats"]').contains(
+            `${remainingSeats}`
+          );
+          cy.get('[data-cy="e2e-admin-used-seats"]').contains(`${usedSeats}`);
+          cy.get('[data-cy="e2e-admin-total-seats"]').contains(`${totalSeats}`);
+
+          // Navigate to users page
+          cy.visit('/admin/users');
+
+          // Set user as admin
+          cy.get('.e2e-user-table')
+            .find('.e2e-user-table-row')
+            .first()
+            .contains('Registered user');
+          cy.get('.e2e-user-table')
+            .find('.e2e-user-table-row')
+            .first()
+            .contains(user1Email);
+          cy.get('.e2e-user-table')
+            .find('.e2e-user-table-row')
+            .first()
+            .find('.e2e-more-actions')
+            .click()
+            .parent()
+            .contains('Set as admin')
+            .click();
+
+          if (usedSeats >= totalSeats) {
+            // Verify that user is required to confirm
+            cy.get('[data-cy="e2e-confirm-change-seat-body"]').should('exist');
+            // Confirm setting user to admin user
+            cy.get('[data-cy="e2e-confirm-change-seat-button"]').click();
+
+            // Check that success is shown and close the modal
+            cy.get('[data-cy="e2e-seat-set-success-body"]').should('exist');
+            cy.get('[data-cy="e2e-close-seat-success-button"]').click();
+          }
+
+          // Verify that user is set to admin
+          cy.get('.e2e-user-table')
+            .find('.e2e-user-table-row')
+            .first()
+            .contains('Platform admin');
+          cy.get('.e2e-user-table')
+            .find('.e2e-user-table-row')
+            .first()
+            .contains(user1Email);
+          cy.get('[data-cy="e2e-admin-and-moderator-count"]').contains(
+            `${adminAndmoderatorsCount + 1}`
+          );
+
+          cy.apiGetAppConfiguration().then((newAppConfigurationResponse) => {
+            // Since we set the user as admin, we need to update the used seats
+            usedSeats = usedSeats + 1;
```

Comment:
Still on approach: Cypress has dangling data which is a good thing for the most part but that also means that there are things we don't have full control on. The variable `userSeats` is used in the tests to compare to the actual value displayed in the UI. We use that value because we cannot rely on the number of seats at any one point but we can rely on the calculation. The reason we cannot rely on the seats is that previous tests in Cypress could set different roles that could change what we are testing for. This gives us predictability and more resilient tests. I prefer to explain this on call but leaving here as-well for anyone else that might need context